backend/src/util/hash.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt::{self, Debug};
+
+use serde::de::{self, Deserialize, Deserializer, Unexpected, Visitor};
+
+const HASH_BYTES: usize = 32;
+
+/// Newtype warpper for 32-byte hash values, implementing readable `Debug` and `serde::Deserialize`.
```

Comment:
"readable", how so?

Reply:
I think the `readable` means formatted as a `hex string` instead of raw bytes ^^